### PR TITLE
Add self signed ssl to test server and test netty client

### PR DESCRIPTION
**Pull request by tylerbenson**
Update netty client instrumentation to accurately capture if https.

### CODE REVIEW

File: dd-java-agent/instrumentation/netty-4.0/src/main/java/datadog/trace/instrumentation/netty40/client/HttpClientRequestTracingHandler.java [hunk 1]
```diff
@@ -39,13 +40,15 @@ public void write(final ChannelHandlerContext ctx, final Object msg, final Chann
     final HttpRequest request = (HttpRequest) msg;
 
     ctx.channel().attr(AttributeKeys.CLIENT_PARENT_ATTRIBUTE_KEY).set(activeSpan());
+    boolean isSecure = ctx.pipeline().get("ssl") != null;
+    NettyHttpClientDecorator decorate = isSecure ? DECORATE_SECURE : DECORATE;
```
Comment (richardstartin) [inline dd-java-agent/instrumentation/netty-4.0/src/main/java/datadog/trace/instrumentation/netty40/client/HttpClientRequestTracingHandler.java:44]: Can we make this a noun not a verb? I've always found this convention a bit strange to read

File: dd-java-agent/testing/src/main/resources/datadog/trace/agent/test/server/http/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+Following https://medium.com/vividcode/enable-https-support-with-self-signed-certificate-for-embedded-jetty-9-d3a86f83e9d9
```
Comment (richardstartin) [inline dd-java-agent/testing/src/main/resources/datadog/trace/agent/test/server/http/README.md:1]: When does it expire?

Comment (richardstartin) [reply dd-java-agent/testing/src/main/resources/datadog/trace/agent/test/server/http/README.md:1]: 100 years?

**Review by richardstartin — APPROVED** [review]
👍🏻

File: dd-java-agent/testing/src/main/resources/datadog/trace/agent/test/server/http/README.md (hunk 2 above)
Comment (tylerbenson) [reply dd-java-agent/testing/src/main/resources/datadog/trace/agent/test/server/http/README.md:1]: yeah... `36500` days. Open to other values if that's a concern.

File: dd-java-agent/instrumentation/netty-4.0/src/main/java/datadog/trace/instrumentation/netty40/client/HttpClientRequestTracingHandler.java (hunk 1 above)
Comment (tylerbenson) [reply dd-java-agent/instrumentation/netty-4.0/src/main/java/datadog/trace/instrumentation/netty40/client/HttpClientRequestTracingHandler.java:44]: I'd prefer to keep it consistent for now.  If we want to change it, that should be done everywhere.

Comment (richardstartin) [reply dd-java-agent/instrumentation/netty-4.0/src/main/java/datadog/trace/instrumentation/netty40/client/HttpClientRequestTracingHandler.java:44]: I'm not sure, changing the names in one go is a cross cutting (i.e. disruptive) change. This is a new local variable name and doesn't need to be consistent with anything.

**Review by bantonsson — APPROVED** [review]

File: dd-java-agent/instrumentation/netty-4.0/src/main/java/datadog/trace/instrumentation/netty40/client/HttpClientRequestTracingHandler.java (hunk 1 above)
Comment (tylerbenson) [reply dd-java-agent/instrumentation/netty-4.0/src/main/java/datadog/trace/instrumentation/netty40/client/HttpClientRequestTracingHandler.java:44]: If you let me know what you'd prefer, I'll' make a PR that changes it everywhere.

**Pull request merged**